code/modules/power/apc/apc_main.dm

```diff
@@ -615,6 +618,18 @@
 /obj/machinery/power/apc/proc/report()
 	return "[area.name] : [equipment]/[lighting]/[environ] ([lastused_total]) : [cell? cell.percent() : "N/C"] ([charging])"
 
+/obj/machinery/power/apc/proc/grey_tide()
+	SIGNAL_HANDLER
+
+	if(!is_station_level(z))
+		return
+
+	for(var/area_type in GLOB.grey_tide_areas)
+		if(istype(get_area(src), area_type))
```

Comment:
Do the same thing here, only UPDATE_ICON and `return` when you succeed.

Also all of these should probably be early-continued.

Reply:
Just makes it look nicer, it's nice to enforce your invariants early